### ISSUE TITLE
Error in view

### ISSUE THREAD

**Issue by chiahetcho** [iDebugAll/nextbox-ui-plugin#24]
Hello, since I updated to 2.8.7, I have this error on one of my sites, 

`"GET /plugins/nextbox-ui/site_topology/2/ HTTP/1.1" 500 1748
Internal Server Error: /plugins/nextbox-ui/site_topology/2/
Traceback (most recent call last):
  File "/opt/netbox-2.8.7/venv/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/opt/netbox-2.8.7/venv/lib/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/opt/netbox-2.8.7/venv/lib/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/opt/netbox-2.8.7/venv/lib/python3.6/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/opt/netbox-2.8.7/venv/lib/python3.6/site-packages/django/contrib/auth/mixins.py", line 85, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/opt/netbox-2.8.7/venv/lib/python3.6/site-packages/django/views/generic/base.py", line 97, in dispatch
    return handler(request, *args, **kwargs)
  File "/opt/netbox-2.8.7/venv/lib/python3.6/site-packages/nextbox_ui_plugin/views.py", line 331, in get
    topology_dict, device_roles, multi_cable_connections, device_tags = get_site_topology(site_id)
  File "/opt/netbox-2.8.7/venv/lib/python3.6/site-packages/nextbox_ui_plugin/views.py", line 304, in get_site_topology
    cable_path, endpoint = link.termination_a.trace()
ValueError: too many values to unpack (expected 2)`

I didn't change anything and I am not sure if it is the right place to post this,

Thanks

EDIT: By changing line 304 of view.py by this :
`cable_path = link.termination_a.trace()`
I don't have anymore issues

**Comment by iDebugAll** [comment]
Thank you for you bug report and analysis.

> cable_path = link.termination_a.trace()
I don't have anymore issues

It seems the trace method was changed in that Netbox release.
This method used to return another string value besides the actual cable path trace result in earlier versions.
Suggested fix will work for 2.8.7+ but I have to consider some backward compatibility for 2.8.0--2.8.6 releases for now.

**Comment by astruwe** [comment]
Hello @iDebugAll and @chiahetcho ,

I've got the same error after updating NetBox to the latest release 2.8.7 and tried using the recommended fix.

Unfortunately a restart of NetBox services didn't help and the error is still the same.

Did I miss something?

**Comment by jessiebryan** [comment]
Also having this issue - using 2.8.7

**Comment by chiahetcho** [comment]
@astruwe Did you "recompile" the plugin after changing that : 

```bash
source /opt/netbox/venv/bin/activate
cd netbox-ui-plugin/
pip3 install . --upgrade
systemctl restart netbox
```

It did the trick for me !

**Comment by iDebugAll** [comment]
I have published a new v0.6.4 release to fix this issue. It should work with all available Netbox versions.

The issue was caused by the change in Netbox's cable trace function in netbox/netbox/dcim/models/device_components.py (was changed in [56ec4a](https://github.com/netbox-community/netbox/commit/56ec4a6360772c5a404d34bedb4b767787267454)):

Before:

>         Return two items: the traceable portion of a cable path, and the termination points where it splits (if any).
>         This occurs when the trace is initiated from a midpoint along a path which traverses a RearPort. In cases where
>         the originating endpoint is unknown, it is not possible to know which corresponding FrontPort to follow.

After:

>         Return three items: the traceable portion of a cable path, the termination points where it splits (if any), and
>         the remaining positions on the position stack (if any). Splits occur when the trace is initiated from a midpoint
>         along a path which traverses a RearPort. In cases where the originating endpoint is unknown, it is not possible
>         to know which corresponding FrontPort to follow. Remaining positions occur when tracing a path that traverses
>         a FrontPort without traversing a RearPort again.

Cable path is the first returned item in both cases. cable_path = link.termination_a.trace() will not work with multi-cable path scenarios. Cable path has to be extracted from the result like this:

```
        trace_result = link.termination_a.trace()
        if not trace_result:
            continue
        cable_path, *ignore = trace_result
```

Please update the plugin and verify if it helped.

**Comment by astruwe** [comment]
@iDebugAll 

Error solved! Thank you very much :).
